src/utils/runtime/src/logger.rs

```diff
@@ -53,6 +55,27 @@ impl Default for LoggerSettings {
 }
 
 impl LoggerSettings {
+    /// Create a new logger settings from the given command-line options.
+    ///
+    /// If env var `RW_TRACING_ENDPOINT` is not set, the meta address will be used
+    /// as the default tracing endpoint, which means that the embedded tracing
+    /// collector will be used.
+    pub fn from_opts<O: risingwave_common::opts::Opts>(opts: &O) -> Self {
+        let mut settings = Self::new(O::name());
+        if settings.tracing_endpoint.is_none() // no explicit env var is set
+            && let Some(addr) = opts.meta_addr().exactly_one() // meta address is valid
+            && !Deployment::current().is_ci()
```

Comment:
I think we need an additional option to let us disable tracing completely, just in case.